cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/SlowBrokerFinder.java

```diff
@@ -0,0 +1,246 @@
+import static com.linkedin.kafka.cruisecontrol.metricsreporter.metric.RawMetricType.BROKER_LOG_FLUSH_TIME_MS_999TH;
+
+
+/**
+ * This class will check whether there is broker performance degradation (i.e slow broker) from collected broker metrics.
+ *
+ * Slow brokers are identified by calculating a derived broker metric
+ * {@code BROKER_LOG_FLUSH_TIME_MS_999TH / (ALL_TOPIC_BYTES_IN + ALL_TOPIC_REPLICATION_BYTES_IN) for each broker and then
+ * checking in two ways.
+ * <ul>
+ *   <li>Comparing the latest against broker's own history.</li>
+ *   <li>Comparing the latest against the last of rest brokers in cluster.</li>
+ * </ul>
+ *
+ * If certain broker's metric value is abnormally high(out of the normal range based on historical value/cluster average),
+ * it is marked as a slow broker suspect by the finder. Then if this suspect broker's derived metric anomaly persists
+ * for some time, it is confirmed to be a slow broker and the finder will report a {@link SlowBrokers}} anomaly
+ * with broker demotion as self-healing proposal. If the metric anomaly still persists for an extended time, the finder
+ * will eventually report another {@link SlowBrokers}} anomaly with broker removal as self-healing proposal.
+ *
+ * Note if there are too many brokers confirmed as slow broker in the same run, the finder will report the {@link SlowBrokers}
+ * anomaly as unfixable. Because this often indicates some serious issue in the cluster and probably requires administrator's
+ * intervention to decide the right remediation strategy.
+ */
+public class SlowBrokerFinder implements MetricAnomalyFinder<BrokerEntity> {
+  private static final Logger LOG = LoggerFactory.getLogger(SlowBrokerFinder.class);
+  public static final String SELF_HEALING_SLOW_BROKERS_REMOVAL_ENABLED_CONFIG = "self.healing.slow.brokers.removal.enabled";
+  public static final String REMOVE_SLOW_BROKERS_CONFIG = "remove.slow.brokers";
+  // Todo: make following configs configurable.
+  // The percentile threshold used to compare latest metric value against historical value.
+  private static final double HISTORY_METRIC_PERCENTILE_THRESHOLD = 90.0;
+  // The margin used to compare latest metric value against historical value.
+  private static final double HISTORY_METRIC_MARGIN = 3.0;
+  // The percentile threshold used to compare last metric value again peers' latest value.
+  private static final double PEER_METRIC_PERCENTILE_THRESHOLD = 50.0;
+  // The margin used to compare last metric value again peers' latest value.
+  private static final double PEER_METRIC_MARGIN = 5.0;
+  // The score threshold to trigger a demotion for slow broker.
+  private static final int SLOW_BROKER_DEMOTION_SCORE = 5;
+  // The score threshold to trigger a removal for slow broker.
+  private static final int SLOW_BROKER_DECOMMISSION_SCORE = 50;
+  // The maximal ratio of slow brokers in the cluster to trigger self-healing operation.
```

Comment:
Looks like this ratio Is not for the `cluster`, but only for the brokers that receive bytes during the sampling period. Is this intentional -- can we update the doc?